File: src/main/java/me/andre111/d20server/command/TriggerCommand.java
package me.andre111.d20server.command;

import me.andre111.d20common.model.profile.Profile;
import me.andre111.d20common.scripting.ScriptException;
import me.andre111.d20server.service.ChatService;

public class TriggerCommand extends Command {
	public TriggerCommand(String name, String[] aliases) {
		super(name, aliases);
	}

	@Override
	public void execute(Profile profile, String arguments) {
		try {
			String[] split = arguments.split(" ");
			if(split.length != 2) throw new ScriptException("Wrong argument count: <messageid> <contentid>");
			
			long messageID = Long.parseLong(split[0]);
			long contentID = Long.parseLong(split[1]);
			
			ChatService.triggerContent(profile, messageID, contentID);
		} catch (ScriptException | NumberFormatException e) {
			ChatService.appendError(profile, "Could execute trigger:", e.getMessage());
		}
	}
}

File: core/server/command/trigger-command.js
// @ts-check
import { I18N } from '../../common/util/i18n.js';
import { splitArguments } from '../../common/util/stringutil.js';
import { ChatService } from '../service/chat-service.js';
import { Command } from './command.js';

export class TriggerCommand extends Command {
    constructor(name, aliases) {
        super(name, aliases, false);
    }

    execute(profile, args) {
        const split = splitArguments(args);
        if (split.length != 2) throw new Error(I18N.get('commands.error.arguments', 'Wrong argument count: %0', '<messageid> <contentid>'));

        const messageID = Number(split[0]);
        const contentID = Number(split[1]);

        ChatService.triggerContent(profile, messageID, contentID);
    }
}
